drivers/DeepSmart/deepsmart/src/deepsmart/wisers.lua

```diff
@@ -0,0 +1,834 @@
+      end
+      -- find hub dev
+      local id = Wisers.idmap[dev_id]
+      if (id == nil) then
+        log.warn('dev '..dev_id..' is not in hub idmap')
+        goto dpretry
+      end
+      log.trace('get dev '..dev_id..' hub devid '..id)
+      local dev = Wisers.driver:get_device_info(id)
+      if (dev == nil) then
+        log.warn('dev '..id..' is not in hub')
+        goto dpretry
+      end
+      if (devtype ~= nil and addrtype ~= nil) then
+        log.trace('dev '..dev_id..' pid '..device.productId..' convert to devtype '..devtype..' addrtype '..addrtype)
+        if (devtype == config.ENUM.AC) then
+          if (addrtype == 0) then
+            local on_off = 'off'
+            if (value ~= 0) then
+              on_off = 'on'
+            end
+            Wisers.driver:set_switch(dev, on_off)
+          elseif (addrtype == 1) then
+            local mode = wiser.dpenum:get_dev_val_by_pid_val(device.productId, dpid, value)
+            Wisers.driver:ac_report(dev, nil, mode, nil,nil, nil,nil)
+          elseif (addrtype == 2) then
+            local fan = wiser.dpenum:get_dev_val_by_pid_val(device.productId, dpid, value)
+            Wisers.driver:ac_report(dev, nil,nil,fan,nil, nil,nil)
+          elseif (addrtype == 3) then
+            Wisers.driver:ac_report(dev, nil,nil,nil,value/100, nil,nil)
+          elseif (addrtype == 4) then
+            Wisers.driver:ac_report(dev, nil,nil,nil,nil, value/100,nil)
+          end
+        elseif (devtype == config.ENUM.HEATER) then
+          if (addrtype == 0) then
+            local on_off = 'off'
+            if (value ~= 0) then
+              on_off = 'on'
+            end
+            Wisers.driver:heater_report(dev, on_off, nil,nil,nil)
+          elseif (addrtype == 2) then
+            Wisers.driver:heater_report(dev, nil,value/100,nil,nil)
+          elseif (addrtype == 3) then
+            Wisers.driver:heater_report(dev, nil,nil,value/100,nil)
+          end
+        elseif (devtype == config.ENUM.NEWFAN) then
+          if (addrtype == 0) then
+            local on_off = 'off'
+            if (value ~= 0) then
+              on_off = 'on'
+            end
+            Wisers.driver:set_switch(dev, on_off)
+          elseif (addrtype == 1) then
+            local fan = wiser.dpenum:get_dev_val_by_pid_val(device.productId, dpid, value)
+            Wisers.driver:ac_report(dev, nil,nil,fan,nil, nil,nil)
+          end
+        end
+      else
+        log.trace('pid '..device.productId..' dpid '..dpid..' convert to devtype addrtype nil')
+      end
+      ::dpretry::
+    end
+    ::knxretry::
+  end
+  return 0
+end
+
+
+--------------
+-- get devtype by device_network_id
+--------------
+function Wisers.get_dev_type(device_network_id)
+  local dev_id,_,_ = Wisers.parse_device_network_id(device_network_id)
+  local wiser,dev = Wisers.get_device_wiser(dev_id)
+  if (wiser == nil) then
+    -- when device is not exist in bridge delete from hub
```

Comment:
old code will delete device when get wiser nil. New code remove the operation.

Reply:
Without this, how will a device removed outside of the SmartThings platform be reflected in the smartthings app?

Reply:
Currently, devices can only be deleted through the SmartThings app